src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -543,12 +546,14 @@ def get_login_credentials(self, resource=None, subscription_id=None, aux_subscri
         identity_type, identity_id = Profile._try_parse_msi_account_name(account)
 
         external_tenants_info = []
-        ext_subs = [aux_sub for aux_sub in (aux_subscriptions or []) if aux_sub != subscription_id]
```

Comment:
> or []) [](start = 61, length = 7)

if both ext_sub and ext_tenants are empty, the original code append sub = self.get_subscription([]), your code removed this logic, please check if it will cause regression

Reply:
In the original code, when aux_subscriptions is empty, ext_subs is [], sub = self.get_subscription (ext_sub) will not be executed.
I understand that the previous code was written this way to prevent aux_subscriptions from having null pointer problems.